Mlem/Views/Tabs/Inbox/InboxView+Feeds.swift

```diff
@@ -0,0 +1,119 @@
+                    await refresh(tracker: personalInboxTracker)
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var moderatorFeedView: some View {
+        Section {
+            switch selectedModTab {
+            case .all:
+                InboxFeedView(tracker: modInboxTracker)
+            case .commentReports:
+                InboxFeedView(tracker: commentReportTracker)
+            case .postReports:
+                InboxFeedView(tracker: postReportTracker)
+            default:
+                InboxFeedView(tracker: modInboxTracker)
+                    .onAppear {
+                        assertionFailure("moderatorFeedView rendered with non-moderator tab!")
+                    }
+            }
+        } header: {
+            BubblePicker(InboxTab.modCases, selected: $selectedModTab, withDividers: [.bottom]) { tab in
+                Text(tab.label)
+            }
+            .background(Color.systemBackground.opacity(scrollToTopAppeared ? 1 : 0))
+            .background(.bar)
+            .animation(.easeOut(duration: 0.2), value: scrollToTopAppeared)
+        }
+        .task {
+            if modInboxTracker.items.isEmpty {
+                // wrap in subtask to view redraws don't cancel load
+                Task(priority: .userInitiated) {
+                    await refresh(tracker: modInboxTracker)
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var adminFeedView: some View {
+        Section {
+            switch selectedModTab {
+            case .all:
+                InboxFeedView(tracker: adminInboxTracker)
+            case .commentReports:
+                InboxFeedView(tracker: commentReportTracker)
+            case .postReports:
+                InboxFeedView(tracker: postReportTracker)
+            case .messageReports:
+                InboxFeedView(tracker: messageReportTracker)
+            case .registrationApplications:
+                InboxFeedView(tracker: registrationApplicationTracker)
+            default:
+                InboxFeedView(tracker: adminInboxTracker)
+                    .onAppear {
+                        assertionFailure("adminFeedView rendered with non-admin tab!")
+                    }
+            }
+        } header: {
+            BubblePicker(InboxTab.adminCases, selected: $selectedModTab, withDividers: [.bottom]) { tab in
+                Text(tab.label)
+            }
+            .background(Color.systemBackground.opacity(scrollToTopAppeared ? 1 : 0))
+            .background(.bar)
+            .animation(.easeOut(duration: 0.2), value: scrollToTopAppeared)
+        }
+        .task {
+            if modInboxTracker.items.isEmpty {
+                // wrap in subtask to view redraws don't cancel load
+                Task(priority: .userInitiated) {
+                    await refresh(tracker: adminInboxTracker)
+                }
+            }
+        }
+    }
```

Comment:
This ViewBuilder is almost identical to the above ViewBuilder. Maybe these should be combined? We can include `.registrationApplications` in the switch statement regardless of whether the tab is accessible or not, and use the ternary operator to choose which tabs are shown in the bubble picker.

Reply:
Good point--I've implemented basically that, although the `BubblePicker` logic is unfortunately a tiny bit less clean because it doesn't handle changing the number of cases well.